### PR TITLE
MAISTRA-710 Create only a single Istio CNI DaemonSet

### PR DESCRIPTION
**Pull request by luksa**

### CODE REVIEW

File: pkg/bootstrap/cni.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
+package bootstrap
+
+import (
```
Comment (rcernich) [inline pkg/bootstrap/cni.go:3]: sort imports?

File: pkg/controller/servicemesh/controlplane/hooks.go [hunk 2]
```diff
@@ -3,6 +3,7 @@ package controlplane
 import (
 	"context"
 	"fmt"
+	"github.com/maistra/istio-operator/pkg/controller/common"
```
Comment (rcernich) [inline pkg/controller/servicemesh/controlplane/hooks.go]: sort imports?

File: deploy/maistra-operator.yaml [hunk 3]
```diff
@@ -450,7 +450,11 @@ spec:
                 fieldRef:
                   fieldPath: metadata.name
             - name: OPERATOR_NAME
-              value: "istio-operator"
+              value: istio-operator
```
Comment (rcernich) [inline deploy/maistra-operator.yaml:453]: need to update the .csv files too.

File: pkg/bootstrap/cni.go (hunk 1 above)
Comment (luksa) [reply pkg/bootstrap/cni.go:3]: Done

File: pkg/controller/servicemesh/controlplane/hooks.go (hunk 2 above)
Comment (luksa) [reply pkg/controller/servicemesh/controlplane/hooks.go]: Done

File: deploy/maistra-operator.yaml (hunk 3 above)
Comment (luksa) [reply deploy/maistra-operator.yaml:453]: Done.

File: manifests/1.0.0/maistra.v1.0.0.clusterserviceversion.yaml [hunk 4]
```diff
@@ -336,7 +336,7 @@ spec:
                   medium: Memory
               containers:
                 - name: istio-operator
-                  image: maistra/istio-operator-ubi8:0.12.0
+                  image: maistra/istio-operator-ubi8:1.0.0
```
Comment (rcernich) [inline manifests/1.0.0/maistra.v1.0.0.clusterserviceversion.yaml]: I think you need to rebase, as there are two of these now, ./manifests-maistra/ and ./manifests-servicemesh/.

Comment (luksa) [reply manifests/1.0.0/maistra.v1.0.0.clusterserviceversion.yaml]: Ah, yes.. thanks... fixed it now.

**Review by rcernich — APPROVED** [review]

File: helm/istio_cni/templates/daemonset.yaml [hunk 5]
```diff
@@ -45,25 +45,25 @@ spec:
...
       # deletion": https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods.
       terminationGracePeriodSeconds: 0
-{{- if .Values.global.priorityClassName }}
-      priorityClassName: "{{ .Values.global.priorityClassName }}"
+{{- if .Values.priorityClassName }}
+      priorityClassName: "{{ .Values.priorityClassName }}"
 {{- end }}
-{{- if .Values.global.imagePullSecrets }}
+{{- if .Values.imagePullSecrets }}
       imagePullSecrets:
```
Comment (jwendell) [inline helm/istio_cni/templates/daemonset.yaml:56]: Can't we fallback to global?

File: helm/istio_cni/templates/daemonset.yaml [hunk 6]
```diff
@@ -45,25 +45,25 @@ spec:
...
       - name: {{ . }}
 {{- end }}
 {{- end }}
       containers:
         # This container installs the Istio CNI binaries
         # and CNI network config file on each node.
         - name: install-cni
-          image: "{{ .Values.global.hub }}/{{ .Values.image }}:{{ .Values.global.tag }}"
-          imagePullPolicy: {{ .Values.global.imagePullPolicy }}
+          image: "{{ .Values.image }}"
```
Comment (jwendell) [inline helm/istio_cni/templates/daemonset.yaml:65]: Why can't we go with global?

File: pkg/controller/common/cni.go [hunk 7]
```diff
@@ -11,6 +12,8 @@ import (
 
 // IsCNIEnabled tells whether this cluster supports CNI or not
 var IsCNIEnabled bool
+var CNIImage string
+var CNIImagePullSecrets []string
```
Comment (jwendell) [inline pkg/controller/common/cni.go:20]: Comment those exported variables?

File: helm/istio_cni/templates/daemonset.yaml [hunk 8]
```diff
@@ -74,21 +74,20 @@ spec:
...
-              value: "{{ .Release.Namespace }}-istio-cni.conf"
+              value: "istio-cni.conf"
             # Name of the kubeconfig file used by CNI agent
             - name: KUBECFG_FILE_NAME
-              value: "{{ .Release.Namespace }}-istio-cni.kubeconfig"
-            # Prefix for the istio-cni binary and install-iptables.sh script
+              value: "istio-cni.kubeconfig"
             - name: CNI_BINARIES_PREFIX
-              value: "{{ .Release.Namespace }}-"
+              value: ""
```
Comment (jwendell) [inline helm/istio_cni/templates/daemonset.yaml:82]: Can we just remove this?

File: helm/istio_cni/templates/daemonset.yaml (hunk 5 above)
Comment (luksa) [reply helm/istio_cni/templates/daemonset.yaml:56]: There is no spoon. :)

There is no global, actually. This is not a sub-chart anymore and is deployed separately from everything else.

File: helm/istio_cni/templates/daemonset.yaml (hunk 6 above)
Comment (luksa) [reply helm/istio_cni/templates/daemonset.yaml:65]: No global + image is configured very differently from all the other images (it should _not_ be configurable by mesh admins, but only by the cluster admin deploying the operator).

File: helm/istio_cni/templates/daemonset.yaml (hunk 8 above)
Comment (luksa) [reply helm/istio_cni/templates/daemonset.yaml:82]: We can, once my Istio CNI PR that fixes this gets merged. Until then, I wanted to make sure these changes work properly.

File: pkg/controller/common/cni.go (hunk 7 above)
Comment (luksa) [reply pkg/controller/common/cni.go:20]: Done.

**Review by jwendell — APPROVED** [review]

**Pull request merged**